### PR TITLE
Clarify and cleanup the test-exemption wording in tree-hygiene.

### PR DESCRIPTION
**Pull request by matanlurey**
Based off a [discussion on Discord](https://discord.com/channels/608014603317936148/608018585025118217/1266501306852446421), but I've also anecdotally had a few people ask this before.

Other than a bit of reformatting, I wanted to clarify that the PR reviewer should still _agree_ an exemption is needed (i.e. not defer this responsibility entirely to our small volunteer staff of reviewers), and that _all_ reviewers should feel empowered to push back, and not wait for a test exemption reviewer to do it. This [matches the guidance in the bot](https://github.com/flutter/cocoon/blob/f1b20b6e4c9f5c1b180425caa166ff4f60001d9e/app_dart/lib/src/service/config.dart#L253-L255).

If this change LGTY I can make a [minor change to the Github bot](https://github.com/flutter/cocoon/blob/f1b20b6e4c9f5c1b180425caa166ff4f60001d9e/app_dart/lib/src/service/config.dart#L239-L255) as well.

### CODE REVIEW

File: docs/contributing/Tree-hygiene.md [hunk 1]
```diff
@@ -84,29 +84,58 @@ _See also: [What should I work on?](What-should-I-work-on.md)_
...
-**In the repositories listed above, a bot will comment on your PR if you need an explicit exemption.** Do not land a PR that has had such a message from the bot unless it has an explicit exemption! The message tells you how to ask for an exemption. Please follow those instructions.
+If a reviewer says a PR should have a test, then it needs a test regardless of the exemptions above.
+
+Feel free to update the bot's logic to catch more cases that should be
+automatically exempt ([in the cocoon repo](https://github.com/reidbaker/cocoon/blob/main/app_dart/lib/src/request_handlers/github/webhook_subscription.dart)).
 
-**If the PR is in a repository that is not listed above, meaning is not supported by the bot, then it is the responsibility of the pull request reviewer to make sure that tests have been added to support the code change.**
+> [!NOTE]
+> PRs adding data-driven fixes require tests that fall under the test_fixes
+> directory, but are _not_ yet recognized by the bot as being tested.
```
Comment (Piinks) [inline docs/contributing/Tree-hygiene.md:107]: Is that so? I thought they were!

File: docs/contributing/Tree-hygiene.md [hunk 2]
```diff
@@ -84,29 +84,58 @@ _See also: [What should I work on?](What-should-I-work-on.md)_
...
+**In the repositories listed above, a bot will comment on your PR if you need an explicit exemption.**
+
+> [!IMPORTANT]
+> If the PR is in a repository that is not listed above, meaning is not
+> supported by the bot, then it is _still_ a responsibility of the pull request
+> reviewer to make sure that tests have been added to support the code change.
+
+### Test Exemptions
+
+A `@test-exemption-reviewer` member on Discord can exempt a PR by commenting
```
Comment (Piinks) [inline docs/contributing/Tree-hygiene.md:128]: Can `Discord` link to the contributing/chat.md page?

File: docs/contributing/Tree-hygiene.md [hunk 3]
```diff
@@ -84,29 +84,58 @@ _See also: [What should I work on?](What-should-I-work-on.md)_
...
+The reviewer of the PR (_not_ the test exemption reviewer) should first:
 
-Consider using the code coverage tools to check that all your new code is covered by tests (see [Test coverage for package:flutter](./testing/Test-coverage-for-package-flutter.md)).
+1. ensure that the PR meets the criteria for exemption
+1. LGTM the PR before requesting the exemption, _or_ clarify why an LGTM was
+   not given.
 
-Feel free to update the bot's logic to catch more cases that should be automatically exempt (it's in the cocoon repo).
+Exemption reviewers are a **small volunteer group** - _all_ reviewers should
+feel empowered to ask for tests.
```
Comment (Piinks) [inline docs/contributing/Tree-hygiene.md:138]: 💯

**Review by Piinks — APPROVED** [review]
LGTM modulo nits

File: docs/contributing/Tree-hygiene.md (hunk 1 above)
Comment (matanlurey) [reply docs/contributing/Tree-hygiene.md:107]: I am just keeping the content from before, but maybe this should be re-evaluated!

File: docs/contributing/Tree-hygiene.md (hunk 2 above)
Comment (matanlurey) [reply docs/contributing/Tree-hygiene.md]: Done!

**Pull request merged**